txircd/ircd.py

```diff
@@ -385,7 +375,150 @@ def reloadModule(self, moduleName):
 		else:
 			deferList.addCallback(lambda result: self.loadModule(moduleName))
 		return deferList
-	
+
+	def verifyConfig(self, config):
+		# Verify the configuration values that are used in the core first.
+
+		# IRCd
+		if "server_name" not in config:
+			raise ConfigValidationError("server_name", "required item not found in configuration file.")
+		if not isinstance(config["server_name"], basestring):
+			raise ConfigValidationError("server_name", "value must be a string")
+		if len(config["server_name"]) > 64:
+			config["server_name"] = config["server_name"][:64]
+			self.logConfigValidationWarning("server_name", "value is too long", config["server_name"])
+		if not re.match(r"^[a-zA-Z0-9.-]+\.[a-zA-Z0-9.-]+$", config["server_name"]):
+			raise ConfigValidationError("server_name", "server name must look like a valid hostname.")
+		if "server_id" in config:
+			if not isinstance(config["server_id"], basestring):
+				raise ConfigValidationError("server_id", "value must be a string")
+			else:
+				config["server_id"] = config["server_id"].upper()
+		else:
+			randFromName = random.Random(config["server_name"])
+			config["server_id"] = randFromName.choice(string.digits)
+			config["server_id"] += randFromName.choice(string.digits + string.uppercase)
+			config["server_id"] += randFromName.choice(string.digits + string.uppercase)
```

Comment:
Does this need to be three lines?
If yes, can we assign a local string and assign that to the dict at the end?